src/fides/api/ops/models/privacy_experience.py

```diff
@@ -165,26 +297,302 @@ def update(self, db: Session, *, data: dict[str, Any]) -> PrivacyExperience:
+    """
+    added_experiences: List[PrivacyExperience] = []
+    updated_experiences: List[PrivacyExperience] = []
+
+    def banner_delivery_update_needed(
+        overlay_component_experience: Optional[PrivacyExperience],
+        overlay_privacy_notices: Query,
+    ) -> bool:
+        """
+        Do we need to update an existing overlay link notice to be delivered by banner instead?
+
+        Any opt-in or notice-ony notices have to be delivered via banner in an overlay.
+        """
+        if (
+            not overlay_component_experience
+            or overlay_component_experience.delivery_mechanism
+            == DeliveryMechanism.banner
+        ):
+            return False
+
+        return bool(
+            overlay_privacy_notices.filter(
+                PrivacyNotice.consent_mechanism.in_(
+                    [ConsentMechanism.notice_only, ConsentMechanism.opt_in]
+                )
+            ).count()
+        )
+
+    def new_experience_needed(
+        existing_experience: Optional[PrivacyExperience], related_notices: Query
+    ) -> bool:
+        """Do we need to create a new experience to match the notices?"""
+        return not existing_experience and bool(related_notices.count())
+
+    for region in affected_regions:
+        (
+            overlay_experience,
+            privacy_center_experience,
+        ) = PrivacyExperience.get_experiences_by_region(db=db, region=region)
+
+        privacy_center_notices: Query = get_privacy_notices_by_region_and_component(
+            db, region, ComponentType.privacy_center
+        )
+        overlay_notices: Query = get_privacy_notices_by_region_and_component(
+            db, region, ComponentType.overlay
+        )
+
+        # See if we need to create a Privacy Center Experience for the Privacy Center Notices
+        if new_experience_needed(
+            existing_experience=privacy_center_experience,
+            related_notices=privacy_center_notices,
+        ):
+            privacy_center_experience = PrivacyExperience.create(
+                db=db,
+                data={
+                    "region": region,
+                    "component": ComponentType.privacy_center,
+                    "delivery_mechanism": DeliveryMechanism.link,
+                },
+            )
+            added_experiences.append(privacy_center_experience)
+
+        # See if we need to update an existing Link Overlay to be delivered by Banner instead
+        if banner_delivery_update_needed(overlay_experience, overlay_notices):
+            assert overlay_experience  # For mypy.  Overlay_experience guaranteed to exist here.
+            overlay_experience.update(
+                db=db,
+                data={
+                    "delivery_mechanism": DeliveryMechanism.banner,
+                    "experience_config_id": None,  # We have to unlink the invalid experience config here if it exists
+                    "experience_config_history_id": None,
```

Comment:
It would be great if we could have a default banner overlay ExperienceConfig here in the system that we could link to on the user's behalf, same for any other experiences we're automatically creating/updating here.

Reply:
agreed. It will be super useful to have some default ExperienceConfigs on the BE in these scenarios. But also, I assume we should still surface to the user when a default has been applied so that they can choose to edit it?

Reply:
yes exactly! I've been mulling over how best to do this (think it should be in the follow-up "defaults" ticket) https://github.com/ethyca/fides/issues/3305.  I was wondering about just a flag on PrivacyExperience, "automatically applied" that I would use if this happened for privacy notices (but not experience configs).  I'm not sure how they "confirm" this though?

Reply:
@eastandwestwind are you okay with us continuing to discuss how to best do this in the followup default ticket?